src/test/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/ClusterMonitorSpec.scala

```diff
@@ -966,4 +969,38 @@ class ClusterMonitorSpec extends TestKit(ActorSystem("leonardotest")) with FlatS
       verify(iamDAO, never()).removeServiceAccountKey(any[GoogleProject], any[WorkbenchEmail], any[ServiceAccountKeyId])
     }
   }
+
+  // Pre:
+  // - cluster exists in the DB with status Error
+  // - dataproc DAO throws exception
+  // Post:
+  // - cluster is not updated in the DB
+  // - monitor actor shuts down
+  // - dataproc DAO should not have been called
+  it should "stop monitoring a cluster in Error status" in isolatedDbTest {
+    val savedErrorCluster = errorCluster.save()
+    savedErrorCluster shouldEqual errorCluster
+
+    val gdDAO = mock[GoogleDataprocDAO]
+    when {
+      gdDAO.getClusterStatus(mockitoEq(creatingCluster.googleProject), mockitoEq(creatingCluster.clusterName))
+    } thenReturn Future.failed(new Exception)
```

Comment:
Do we need the `when { }` block above given we `verify(gdDAO, never())...` and we don't care about the `gdDAO` response but instead that we never called `gdDAO` at the first place? If it's necessary, shouldn't it input `errorCluster...` instead of `creatingCluster...`?

Reply:
You're right it's probably not necessary since we assert that `gdDAO.getClusterStatus` never gets called (and yes, it should be `errorCluster`, nice catch :)).

I think I included it because that's what happens currently -- `gdDAO.getClusterStatus` throws an exception because the project is not active. But with this change it should not get called.